[c10] helpers for runtime c10::alias re-use

dolpm opened the pull request:
Summary: we need these to check whether the input tensor was re-sized/strided between executions when choosing to alias

Test Plan: CI

Reviewed By: henryoier

Differential Revision: D73061676

Skylion007 commented on c10/core/impl/SizesAndStrides.h:
```diff
@@ -50,6 +50,20 @@ class C10_API SizesAndStrides {
+    }
+    return !(
+        isInline()
+            ? memcmp(
```
Use std::memcmp here for better code hygiene

Skylion007 replied:
Use std::memcmp here for better code hygiene

Skylion007 commented on c10/core/impl/SizesAndStrides.h:
```diff
@@ -50,6 +50,20 @@ class C10_API SizesAndStrides {
+        isInline()
+            ? memcmp(
+                  inlineStorage_, other.inlineStorage_, sizeof(inlineStorage_))
+            : memcmp(
```
Same here

dolpm committed:
[c10] helpers for runtime c10::alias re-use (#151361)

Summary:

we need these to check whether the input tensor was re-sized/strided between executions when choosing to alias

Test Plan: CI

Reviewed By: SherlockNoMad, henryoier

Differential Revision: D73061676

Camyll commented:
@pytorchbot merge
